week5/Q1.py

```diff
@@ -0,0 +1,149 @@
+class OrientedGraph:
+	'''
+	Implementation of oriented graph. All of the nodes have unique values.
+
+	Fields:
+		nodes: set of Nodes, all nodes of the graph
+	'''
+	def __init__(self):
+		self.nodes = set()
+
+	def add(self, value, parent_value = None):
+		'''
+		Adds new Node to the OrientedGraph or updates existing ones.
+
+		Args:
+			value: str
+			parent_value: str
+		'''
+		node = None
+		parent = None
+		for inode in self.nodes:
+			if inode.value == value:
+				node = inode
+			if inode.value == parent_value:
+				parent = inode
+
+		if parent_value:
+			if node:
+				node.parents = node.parents.union({parent})
```

Comment:
It is possible for parent_value to be valid but the parent node was not found.
You should decide how to handle this.
Options are:
- `raise ValueError('parent not found: ' + parent_value)`
- insert the `parent_value` into the graph
- ignore the `parent_value` if not found